lib/common/ipc_common.c

```diff
@@ -70,3 +70,39 @@ pcmk__client_type_str(uint64_t client_type)
             return "unknown";
     }
 }
+
+bool
+pcmk__ipc_msg_is_multipart(void *data)
+{
+    pcmk__ipc_header_t *header = data;
+
+    if (!pcmk__valid_ipc_header(header)) {
+        return false;
+    }
+
+    return pcmk_all_flags_set(header->flags, crm_ipc_multipart);
```

Comment:
`pcmk_is_set()` would be the more intuitive call here and below. They do exactly the same thing, `pcmk_is_set()` is just a wrapper for clarity with a single flag.

And could just be
```
return pcmk__valid_ipc_header(header)
       && pcmk_is_set(header->flags, crm_ipc_multipart);
```

Whatever you think is more readable.

Reply:
Having written the code now, I like moving the header check to the callers.  We were actually already doing that everywhere except for one spot, so the checks were redundant.